build_test/lib/src/test_builder.dart

```diff
@@ -119,6 +119,15 @@ Future testBuilder(
     inputIds.add(id);
   });
 
+  var allPackages = reader.assets.keys.map((a) => a.package).toSet();
+  for (var pkg in allPackages) {
+    for (var dir in const ['lib', 'web', 'test']) {
+      var asset = new AssetId(pkg, '$dir/\$$dir\$');
+      reader.cacheBytesAsset(asset, const []);
+      inputIds.add(asset);
```

Comment:
What if we only do this and just remove the line above that adds the asset to the reader? I think that might give the desired behavior.

Reply:
Sure. I'll tackle this before merging.

Reply:
Done.